samples/Net7Worker/EventHubCancellationToken.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+
+namespace Net7Worker
+{
+    public class EventHubCancellationToken
+    {
+        private readonly ILogger _logger;
+
+        public EventHubCancellationToken(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<EventHubCancellationToken>();
+        }
+
+        [Function(nameof(EventHubCancellationToken))]
+        public void Run(
+            [EventHubTrigger("src", Connection = "EventHubConnection")] string[] messages,
+            FunctionContext context,
+            CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("C# EventHub trigger function processing a request.");
+
+            try
+            {
+                foreach (var message in messages)
+                {
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
The sample could simply throw at the beginning of the look (using `ThrowIfCancellationRequested`). You can document the pattern of checking `IsCancellationRequested` for scenarios where additional work/cleanup is required. But if we just to short circuit the loop, throwing would be sufficient.